crates/noirc_evaluator/src/lib.rs

```diff
@@ -40,14 +44,17 @@ pub fn create_circuit(
     is_blackbox_supported: IsBlackBoxSupported,
     enable_logging: bool,
     show_output: bool,
-) -> Result<Circuit, RuntimeError> {
+) -> Result<(Circuit, Abi), RuntimeError> {
     let mut evaluator = Evaluator::new();
 
     // First evaluate the main function
-    evaluator.evaluate_main_alt(program, enable_logging, show_output)?;
+    evaluator.evaluate_main_alt(program.clone(), enable_logging, show_output)?;
```

Comment:
One final nit: Instead of cloning program, can we clone the abi before passing ownership of it to `evaluator.evaluate_main_alt`

This might be optimised away by the compiler, not sure

Reply:
Yeah, I was thinking about this but I preferred the clarity of keeping the abi manipulations together. We can revisit this later imo.